src/components/Floorplan.js

```diff
@@ -21,16 +26,24 @@ export default class Floorplan extends React.Component {
 
     paper.setup(this.refs.canvas);
 
-    const seats: Seat[] = [];
+    this.generateSeats();
+  }
 
-    // Render the seats based on the loaded data.
+  /**
+   *  Render seats on the floorplan based off of the Store.
+   */
+  generateSeats() {
     for (const id in this.props.seats) {
       const seatdata = this.props.seats[id];
       const seat = new Seat(seatdata.x, seatdata.y, seatdata.label);
-      seats.push(seat);
+
+      seat.onSelect = () => this.handleSelectSeat(id);
     }
+  }
 
-    console.log('Floorplan initialized.');
+  handleSelectSeat(id: string) {
```

Comment:
I don't agree on this one, 
`handleSelectedSeat` means it's already selected. The action of the selection should happen INSIDE this function and not before.